File: src/main/java/victor/kata/parking/ParkingBuilder.java
package victor.kata.parking;

import java.util.ArrayList;
import java.util.List;

/**
 * Builder class to get a parking instance
 */
public class ParkingBuilder {

    private int size;
    private final List<Integer> pedestrianExits = new ArrayList<>();
    private final List<Integer> disabledBays = new ArrayList<>();

    public ParkingBuilder withSquareSize(final int size) {
        this.size = size;
        return this;
    }

    public ParkingBuilder withPedestrianExit(final int pedestrianExit) {
        this.pedestrianExits.add(pedestrianExit);
        return this;
    }

    public ParkingBuilder withDisabledBay(final int disabledBay) {
        this.disabledBays.add(disabledBay);
        return this;
    }

    public Parking build() {
        return new Parking(size, pedestrianExits, disabledBays);
    }
}

File: src/main/java/victor/kata/parking/ParkingBuilder.kt
package victor.kata.parking

/**
 * Builder class to get a parking instance
 */
class ParkingBuilder {

    private var size: Int? = null
    private val pedestrianExits: MutableSet<Int> = mutableSetOf()
    private val disabledBays: MutableSet<Int> = mutableSetOf()

    fun withSquareSize(size: Int): ParkingBuilder {
        this.size = size
        return this
    }

    fun withPedestrianExit(pedestrianExitIndex: Int): ParkingBuilder {
        pedestrianExits += pedestrianExitIndex
        return this
    }

    fun withDisabledBay(disabledBayIndex: Int): ParkingBuilder {
        disabledBays += disabledBayIndex
        return this
    }

    fun build(): Parking {
        return Parking(requireNotNull(size), pedestrianExits, disabledBays)
    }
}